sqlmesh/dbt/context.py

```diff
@@ -281,3 +281,13 @@ def create_relation(
 
 
 SQLMESH_DBT_PACKAGE = "sqlmesh.dbt"
+
+TARGET_TYPE_TO_DIALECT: t.Dict[str, str] = {
+    "databricks": "databricks",
+    "duckdb": "duckdb",
+    "postgres": "postgres",
+    "redshift": "redshift",
+    "snowflake": "snowflake",
+    "bigquery": "bigquery",
+    "sqlserver": "tsql",
+}
```

Comment:
What do you think about calling this `TARGET_TYPE_TO_DIALECT_OVERRIDE` and then only having `sqlserver` -> `tsql` in it. Calls do a get and then default to the `target.type` if no match. Avoids having to make sure we have entries for target types that directly map to a dialect (which is the common case).